consensus/src/quorum_store/utils.rs

```diff
@@ -300,27 +396,24 @@ impl ProofQueue {
         while !iters.is_empty() {
             iters.shuffle(&mut thread_rng());
             iters.retain_mut(|iter| {
-                if full {
-                    return false;
-                }
                 if let Some((sort_key, batch)) = iter.next() {
                     if excluded_batches.contains(batch) {
                         excluded_txns += batch.num_txns();
                     } else if let Some(Some((proof, insertion_time))) =
                         self.batch_to_proof.get(&sort_key.batch_key)
                     {
-                        cur_bytes += batch.num_bytes();
-                        cur_txns += batch.num_txns();
-                        if cur_bytes > max_bytes || cur_txns > max_txns {
-                            // Exceeded the limit for requested bytes or number of transactions.
+                        if cur_bytes + batch.num_bytes() > max_bytes
+                            || cur_txns + batch.num_txns() > max_txns
+                        {
                             full = true;
                             return false;
                         }
+                        cur_bytes += batch.num_bytes();
```

Comment:
They are actual changes. The previous logic seems to have a bug. So, I fixed it here.

Reply:
Consider this example.
cur_txns = 1700
Suppose peer_1's next batch contains 250 transactions, we first update cur_txns to 1950. As it exceeds the max_txns (1900) limit, we don't include any more batches from any other others.

Reply:
With my change,
   cur_txns = 1700
Suppose peer_1's next batch contains 250 transactions. As 1700+250 exceeds the max_txns (1900) limit, we don't include any more batches from peer_1. We now move to peer_2.
Suppose peer_2's next batch contains 150 transactions. cur_txns is now 1700. So, we include this batch.

Reply:
I see, makes sense. Can you please include a comment on this change to the PR summary? That can help future reviewers.

In the future, this might be better as a separate PR.

Reply:
I have a separate PR for this as well :-) 
https://github.com/aptos-labs/aptos-core/pull/13716